mgqpy/operators/eq_ne_not.py

```diff
@@ -10,9 +12,9 @@ def _match_eq(doc, path: List[str], ov) -> bool:
             return True
 
         if isinstance(ov, re.Pattern) and isinstance(doc, str):
-            if ov.search(doc):
-                return True
+            return bool(ov.search(doc))
```

Comment:
regarding this, if one is a pattern and the other side is not a string, but for instance an UUID - we could coerce to string, yes, could technically be helpful in the good old ObjectId as a/b bucket flag scenario, or some more advanced filter logic. But on the other hand I haven't thought through possible failure modes, so maybe best to just keep it as is for now.